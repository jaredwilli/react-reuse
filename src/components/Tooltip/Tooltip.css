.tooltip {
    display: inline-block;
    position: relative;
}
.tooltip .tooltip-label {
    color: #3566A0;
    display: inline-block;
}
.tooltip .tooltip-label:hover,
.tooltip .tooltip-content a {
    cursor: pointer;
}
.tooltip .tooltip-label svg {
    vertical-align: middle;
    margin-left: 5px;
}
.underlay {
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
}
.tooltip .tooltip-content {
    background: #979797;
    box-shadow: 0 1px 4px 0 rgba(0, 0, 0, 0.19);
    border-radius: 4px;
    color: #fff;
    font-size: 14px;
    line-height: 20px;
    position: absolute;
    left: 93.4%;
    top: 100%;
    min-width: 360px;
    margin-top: 1em;
    padding: 33px 24px;
    transform: translateX(-50%);
    opacity: 0.94;
    z-index: 1000;
}
.tooltip .tooltip-content:after {
	border: solid transparent;
	border-bottom-color: #979797;
	border-width: 10px;
	position: absolute;
	bottom: 100%;
	left: 50%;
	content: ' ';
	height: 0;
	width: 0;
    margin-left: -10px;
	pointer-events: none;
}
.tooltip.is-visible .tooltip-content {
    transform: translateY(0) translateX(-50%);
    opacity: 1;
    visibility: visible;
}
.tooltip.is-hidden .tooltip-content,
.tooltip.is-hidden .underlay {
    transform: translateY(100%) translateX(-50%);
    opacity: 0;
    visibility: hidden;
}
.tooltip.is-visible .underlay {
    opacity: 1;
    visibility: visible;
}
