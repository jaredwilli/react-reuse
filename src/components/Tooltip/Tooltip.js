import React from 'react';
import PropTypes from 'prop-types';

import QuestionMark from '../QuestionMark/QuestionMark';

import './Tooltip.css';

/** Tooltip component that takes a label for the text to tooltip and children for the content of the tooltip itself. */
class Tooltip extends React.Component {
    constructor(props) {
        super(props);

        this.state = {
            isVisible: false
        };
    }

    componentDidMount() {
        document.addEventListener('mousedown', this.onMouseOut);
    }

    componentWillUnmount() {
        document.removeEventListener('mousedown', this.onMouseOut);
    }

    onMouseOver = () => {
        this.setState({
            isVisible: true
        });
    }

    onMouseOut = () => {
        this.setState({
            isVisible: false
        });
    }

    render() {
        const isVisible = this.state.isVisible ? 'is-visible' : 'is-hidden';
        const className = `tooltip ${isVisible}`;
        const { label, click } = this.props;

        if (click) {
            return (
                <div
                    className={className}
                    onClick={() => this.onMouseOver()}
                >
                    <div className="tooltip-label">
                        {label}

                        <QuestionMark />
                    </div>

                    <div className="tooltip-content">
                        {this.props.children}
                    </div>
                </div>
            );
        }

        return (
            <div
                className={className}
                onMouseEnter={() => this.onMouseOver()}
                onMouseOut={() => this.onMouseOut()}
            >
                <div className="tooltip-label">
                    {label}

                    <QuestionMark />
                </div>

                <div className="tooltip-content">
                    {this.props.children}
                </div>
            </div>
        );
    }
}

Tooltip.defaultProps = {
    click: false
};

Tooltip.propTypes = {
    /** The text to show for the tooltip that you hover */
    label: PropTypes.string.isRequired,

    /** If the tooltip should be activated on click this is true */
    click: PropTypes.bool
};

export default Tooltip;
