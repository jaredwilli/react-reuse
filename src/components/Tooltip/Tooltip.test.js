import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Tooltip from './Tooltip';

describe('Tooltip', () => {
    it('renders the component correctly', () => {
        const tree = renderer.create(<Tooltip label="M.Gemi Fit Expert">Test</Tooltip>).toJSON();
        expect(tree).toMatchSnapshot();
    });
});
