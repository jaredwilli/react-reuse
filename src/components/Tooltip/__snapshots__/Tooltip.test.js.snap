// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tooltip renders the component correctly 1`] = `
<div
  className="tooltip is-hidden"
  onMouseEnter={[Function]}
  onMouseOut={[Function]}
>
  <div
    className="tooltip-label"
  >
    M.Gemi Fit Expert
    <svg
      height={19}
      viewBox="0 0 19 19"
      width={19}
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fill="none"
        fillRule="evenodd"
        transform="translate(1 1)"
      >
        <circle
          cx="8.5"
          cy="8.5"
          r="8.5"
          stroke="#3566A0"
        />
        <path
          d="M7.299 5.567c.263-.603.99-1.175 1.732-1.175.82 0 1.5.541 1.5 1.283 0 1.64-3.449 2.088-3.449 4.145 0 .556.202 1.051.65 1.546l1.067-.928c-.155-.17-.247-.34-.247-.541C8.552 8.768 12 8.257 12 5.66 12 4.237 10.825 3 9.03 3 7.795 3 6.48 3.773 6 4.948l1.299.619zm-.773 7.485c0 .603.48 1.082 1.082 1.082.603 0 1.083-.48 1.083-1.082 0-.604-.48-1.083-1.083-1.083s-1.082.48-1.082 1.083z"
          fill="#3566A0"
        />
      </g>
    </svg>
  </div>
  <div
    className="tooltip-content"
  >
    Test
  </div>
</div>
`;
